src/main.c

```diff
@@ -781,20 +783,34 @@ static void open_activated(GSimpleAction* action,
         "_Cancel", GTK_RESPONSE_CANCEL,
         "_Open", GTK_RESPONSE_ACCEPT,
         NULL);
-
-    strcpy(splits_path, win->data_path);
-    strcat(splits_path, "/splits");
-    if (stat(splits_path, &st) == -1) {
-        mkdir(splits_path, 0700);
+    filter = gtk_file_filter_new();
+    gtk_file_filter_add_pattern(filter, "*.json");
+    gtk_file_filter_set_name(filter, "LibreSplit JSON Split Files");
+    gtk_file_chooser_add_filter(GTK_FILE_CHOOSER(dialog), filter);
+
+    if (json_string_value(get_setting_value("libresplit", "last_split_folder")) != NULL) {
```

Comment:
This can be saved into a variable for it then to be used inside the if and the strcpy without checking it twice